src/version.ts

```diff
@@ -0,0 +1,318 @@
+  { platform: "freebsd", arch: "x86_64", ext: "tar.gz" },
+]
+const DEFAULT_NIGHTLY_PLATFORM = {
+  platform: "linux",
+  arch: "x86_64",
+  ext: "tar.gz",
+}
+
+type Download = {
+  url: string
+  triplet?: string
+  kind: string
+  arch: string
+  asc?: string
+  sha256?: string
+  size: number
+  version: string
+  os: string
+  extension: string
+}
+
+type JuliaVersionsJson = {
+  [key: string]: {
+    files: Array<Download>
+    stable: boolean
+  }
+}
+
+type NightlyPlatform = {
+  platform: string
+  arch: string
+  suffix?: string
+  ext: string
+}
+
+/**
+ * Determine a specific Julia version for each version specifier.
+ *
+ * @param versionSpecifiers: A list of version specifiers. See the README for
+ * details on the syntax.
+ * @param project: The Julia project directory or file to use when determining
+ * Julia compatibility with a project.
+ * @param options: The `ifMissing` option controls the behavior of this
+ * function when a version specifier cannot be resolved.
+ * @returns A list of resolved versions
+ * @throws Error if a version specifier doesn't resolve to any available
+ * Julia release
+ */
+export async function resolveVersions(
+  versionSpecifiers: Array<string>,
+  project: string = ".",
+  options?: { ifMissing: string }
+): Promise<Array<string | null>> {
+  // Determine the Julia compat ranges as specified by the Project.toml only
+  // for aliases that require them.
+  let juliaCompatRange: string = ""
+  if (versionSpecifiers.includes("min")) {
+    const juliaProjectFile = getJuliaProjectFile(project)
+    const juliaProjectToml = toml.parse(
+      fs.readFileSync(juliaProjectFile).toString()
+    )
+
+    // Extract the compat range for the "julia" entry and convert it to the
+    // node semver range syntax.
+    juliaCompatRange = getJuliaCompatRange(juliaProjectToml)
+    core.debug(`Julia project compatibility range: ${juliaCompatRange}`)
+  }
+
+  const availableVersions = Object.keys(await fetchJuliaVersionsJson())
+
+  const resolvedVersions = new Array<string | null>()
+  for (const versionSpecifier of versionSpecifiers) {
+    let resolvedVersion: string | null
+
+    // Nightlies are not included in the versions.json file
+    const nightlyMatch = /^(?:(\d+\.\d+)-)?nightly$/.exec(versionSpecifier)
+    if (nightlyMatch && nightlyMatch[1]) {
+      const url = getNightlyUrl(DEFAULT_NIGHTLY_PLATFORM, nightlyMatch[1])
+      resolvedVersion = (await urlExists(url)) ? versionSpecifier : null
+    } else if (nightlyMatch) {
+      // Skip URL check for "nightly" as it should always be available
+      resolvedVersion = versionSpecifier
+    } else {
+      resolvedVersion = resolveVersion(
+        versionSpecifier,
+        availableVersions,
+        juliaCompatRange
+      )
+    }
+
+    core.debug(`${versionSpecifier} -> ${resolvedVersion}`)
+
+    if (resolvedVersion) {
+      resolvedVersions.push(resolvedVersion)
+    } else if (options?.ifMissing === "warn") {
```

Comment:
[SuperNit] Since `warn` is the default behaviour, I would make that the base `else`, and `else if` the case for `ifMissing === "error"`

Reply:
Changing this with the current setup would mean that an invalid `if-missing` input would result in a warning instead of an error. I'll update this value to make better use of types to catch this kind of thing earlier